presto-hive/src/main/java/com/facebook/presto/hive/parquet/reader/ParquetReader.java

```diff
@@ -124,7 +133,57 @@ private boolean advanceToNextRowGroup()
         return true;
     }
 
-    public Block readStruct(Type type, List<String> path)
+    public Block readArray(Type type, List<String> path, List<Integer> positions)
+            throws IOException
+    {
+        List<Type> parameters = type.getTypeParameters();
+        checkArgument(parameters.size() == 1, "Array must has 1 parameter, this array has %d parameters", parameters.size());
+        List<Integer> lengths = new ArrayList<>();
+        path.add(ARRAY_TYPE_NAME);
+        Block block = readBlock(ARRAY_ELEMENT_NAME, parameters.get(0), path, lengths, positions);
+        path.remove(ARRAY_TYPE_NAME);
+
+        if (lengths.size() == 0) {
+            return RunLengthEncodedBlock.create(parameters.get(0), null, batchSize);
+        }
+
+        int[] offsets = new int[batchSize + 1];
+        for (int i = 1; i < offsets.length; i++) {
+            offsets[i] = offsets[i - 1] + lengths.get(i - 1);
+        }
+        return new ArrayBlock(batchSize, new boolean[batchSize], offsets, block);
```

Comment:
Two comments here regarding the `isNull` array passed to the `ArrayBlock` constructor:
- Why don't we populate this array properly instead of passing a brand new one with each call? Are we sure that there is no other method that depends on this array to be valid?
- We are creating these `1K` `isNull` boolean arrays and throw them away with each call, although they are small when we have a large number of rows this will produce some garbage on the heap. If we really don't need them we can have a static final boolean array that's init once instead of creating a new one all the time.